augur/util_support/date_disambiguator.py

```diff
@@ -149,3 +142,24 @@ def assert_only_less_significant_uncertainty(self):
                 raise InvalidDate(self.uncertain_date,
                     "Month contains uncertainty, so day must also be uncertain."
                 )
+
+    def set_bounds(self, min_max_year):
+        """Set the exact day bounds."""
+        if len(min_max_year) > 2:
+            raise AugurError(f"The year bounds '{min_max_year}' must have only one (lower) or two (lower, upper) bounds.")
+
+        # TODO: Allow more granularity than year alone.
+
+        try:
+            self.lower_bound = datetime.date(int(min_max_year[0]), 1, 1)
+        except:
+            raise AugurError(f"'{min_max_year[0]}' is not a valid year.")
+
+        if len(min_max_year) == 2:
+            try:
+                self.upper_bound = datetime.date(int(min_max_year[1]), 12, 31)
+            except:
+                raise AugurError(f"'{min_max_year[1]}' is not a valid year.")
+
+        if self.lower_bound > self.upper_bound:
+            raise AugurError(f"The year bounds '{min_max_year}' must be in the order of (earlier, later).")
```

Comment:
Many things which take an ordered pair will issue a warning instead of an error if they're reversed, and then automatically swap them.

Do you think that's an appropriate UX choice here? Or would it potentially mask larger issues in whatever's calling Augur?

Reply:
It sounds appropriate, at least for the only place this is currently used, `augur refine --year-bounds`, whose help text says:

> specify min or max & min prediction bounds for samples with XX in date

which doesn't indicate an expectation of order (the order is even backwards in this text!). Given the unclear expected order, I'll add the swap here without any warning.

Reply:
Done: 77d8d8b1ba0ee5e988f5bd015fe25fe53a153120

Reply:
Hmm, the auto-swap there is applied even if the user provided only `--year-bounds min` without a max in which case it compares the user-provided min to the default max of the current date. This is probably surprising/undesirable since the user's min would become the max? Maybe best to 1) restrict the auto-swap to when the user provides both min and max, and 2) if they only provide min, either ensure a) its before the default max or b) the default max is adjusted to some other value (but what?). Thoughts?

Reply:
Both of those sound good to me!

One potential caveat that comes to mind with (2) is that when the default is specified at an outer layer, the deeper part may still want/need to know that the value was a default and not user-provided. But we'll see if that's information that needs passing in actuality or not.

Reply:
I updated this PR so that swapping only happens when both values are provided.

Also, I moved the default max logic back to where it originally was, so the behavior is now in line with 2)a) from what you described above:

> 2) if they only provide min, ensure a) its before the default max